portfolio/src/test/java/com/google/sps/CalendarGroupTest.java

```diff
@@ -0,0 +1,317 @@
+    Instant startTime = Instant.now();
+    Instant endTime = startTime.plusSeconds(5000);
+
+    Instant eventOneStart = startTime.plusSeconds(2000);
+    Instant eventOneEnd = eventOneStart.plusSeconds(1000);
+    Instant eventTwoStart = startTime.plusSeconds(1000);
+    Instant eventTwoEnd = eventTwoStart.plusSeconds(3000);
+
+    CalendarEvent eventOne = new CalendarEvent("event one", eventOneStart, eventOneEnd);
+    CalendarEvent eventTwo = new CalendarEvent("event two", eventTwoStart, eventTwoEnd);
+    events.add(eventOne);
+    events.add(eventTwo);
+
+    CalendarGroup eventGroup = new CalendarGroup(events, startTime, endTime);
+    List<TimeRange> freeTimeRanges = eventGroup.getFreeTimeRanges();
+
+    Instant freeOneStart = startTime;
+    Instant freeOneEnd = startTime.plusSeconds(1000);
+    Instant freeTwoStart = startTime.plusSeconds(4000);
+    Instant freeTwoEnd = endTime;
+    TimeRange freeOne = TimeRange.fromStartEnd(freeOneStart, freeOneEnd);
+    TimeRange freeTwo = TimeRange.fromStartEnd(freeTwoStart, freeTwoEnd);
+
+    Assert.assertTrue(TimeRange.equals(freeOne, freeTimeRanges.get(0)));
+    Assert.assertTrue(TimeRange.equals(freeTwo, freeTimeRanges.get(1)));
+  }
+
+  /** Tests for scenario when there is no free time. */
+  @Test
+  public void noFreeTime() {
+    // Events:   |--One--||--------Two-------|
+    // Possible: |---------------------------|
+    // Free:
+
+    List<CalendarEvent> events = new ArrayList<CalendarEvent>();
+    Instant startTime = Instant.now();
+    Instant endTime = startTime.plusSeconds(2000);
+
+    Instant eventOneStart = startTime;
+    Instant eventOneEnd = eventOneStart.plusSeconds(1000);
+    Instant eventTwoStart = eventOneEnd;
+    Instant eventTwoEnd = eventTwoStart.plusSeconds(1000);
+
+    CalendarEvent eventOne = new CalendarEvent("event one", eventOneStart, eventOneEnd);
+    CalendarEvent eventTwo = new CalendarEvent("event two", eventTwoStart, eventTwoEnd);
+    events.add(eventOne);
+    events.add(eventTwo);
+
+    CalendarGroup eventGroup = new CalendarGroup(events, startTime, endTime);
+    List<TimeRange> freeTimeRanges = eventGroup.getFreeTimeRanges();
+    Assert.assertTrue(freeTimeRanges.isEmpty());
+  }
+
+  /**
+   * Tests for scenario where there is no calendar event. Entirety of possible time range is the
+   * free time range.
+   */
+  @Test
+  public void allFreeTime() {
+    // Events:
+    // Possible: |---------------------------|
+    // Free:     |---------------------------|
+
+    List<CalendarEvent> events = new ArrayList<CalendarEvent>();
+    Instant startTime = Instant.now();
+    Instant endTime = startTime.plusSeconds(10000);
+    TimeRange targetFreeTime = TimeRange.fromStartEnd(startTime, endTime);
+
+    CalendarGroup eventGroup = new CalendarGroup(events, startTime, endTime);
+    List<TimeRange> freeTimeRanges = eventGroup.getFreeTimeRanges();
+
+    Assert.assertTrue(freeTimeRanges.get(0).equals(targetFreeTime));
```

Comment:
TimeRange implements an equals, so can we compare the Lists instead? Maybe we can use an expected and actual List. For example:
`List<TimeRange> expectedFreeTimeRanges = Arrays.asList(targetFreeTime);`
`List<TimeRange> actualFreeTimeRanges = eventGroup.getFreeTimeRanges();`
`Assert.assertEquals(expectedFreeTimeRanges, actualFreeTimeRanges);`

Reply:
I see, updated!